src/main/java/com/qubit/solution/fenixedu/integration/cgd/services/studentPhoto/CgdStudentPhotoClient.java

```diff
@@ -0,0 +1,157 @@
+            this.category = member.getMemberCategoryCode().getValue();
+            this.memberNumber = member.getMemberNumber().getValue();
+        }
+
+        private Person getPersonFromDocumentId() {
+            Collection<Person> persons = Person.findPersonByDocumentID(this.memberNumber);
+            if (persons.size() > 1) {
+                logger.warn("Too many persons match the member number sent from CGD when using document id number.");
+                return null;
+            } else {
+                return Iterables.getFirst(persons, null);
+            }
+        }
+
+        public Person getPerson() {
+            Person personFromDocumentId = getPersonFromDocumentId();
+            if (isMemberNumberLikeIdDocumentNumber()) {
+                return personFromDocumentId;
+            } else if (isStudent()) {
+                Student student = Student.readStudentByNumber(getMemberAsNumber());
+                if (student != null) {
+                    return student.getPerson();
+                }
+            } else if (isStaff()) {
+                Optional<Employee> employeeOpt = Employee.findByNumber(this.memberNumber);
+                if (employeeOpt.isPresent()) {
+                    return employeeOpt.get().getPerson();
+                }
+                if (personFromDocumentId != null) {
+                    return personFromDocumentId;
+                }
+            }
+            return null;
+        }
+
+        public String getMemberNumber() {
+            return this.memberNumber;
+        }
+
+        public String getCategory() {
+            return category;
+        }
+    }
+
+    public static class PhotoOnCGD {
+        private byte[] photoContent;
+        private String fileName;
+
+        private PhotoOnCGD(byte[] ba, String fileName) {
+            this.photoContent = ba;
+            this.fileName = fileName;
+        }
+
+        public byte[] getPhotoContent() {
+            return photoContent;
+        }
+
+        public String getFileName() {
+            return fileName;
+        }
+
+    }
+
+    public Collection<MemberCGD> getStudentsWithPhotos() {
+        logger.debug("CgdPhotoWsClient.getStudentsWithPhotos()");
+        IStudentPhotoService ws = (IStudentPhotoService) getClient();
+        List<Member> members = ws.getAllStudentsByIES().getMember();
+
+        List<MemberCGD> memberCGDs = new ArrayList<MemberCGD>();
+        for (Member member : members) {
+            memberCGDs.add(new MemberCGD(member));
+        }
+        return memberCGDs;
+    }
+
+    public PhotoOnCGD getPhoto(final MemberCGD member) {
+        return getPhoto(member.getMemberNumber(), member.getCategory());
+    }
+
+    /*
+     * memberCategory parameter is optional. It is used by CGD to disambiguate between members with
+     * the same number and different category.
+     */
+    public PhotoOnCGD getPhoto(final String memberNumber, final String memberCategory) {
```

Comment:
@pmmsa again the PhotoOnCGD versus Optional<PhotoOnCGD> being this a public method (not like the other that was private) I have strong feelings about this being an Optional :-)